### DISCUSSION TITLE
blog/nextjs-storybook-tailwind

### DISCUSSION THREAD

**Discussion by giscus** [theodorusclarence/theodorusclarence.com#228] [General]
# blog/nextjs-storybook-tailwind

Complete configuration and setup for Storybook with Next.js and Tailwind CSS

https://theodorusclarence.com/blog/nextjs-storybook-tailwind

**Comment by awthwathje** [comment]
Nice guide, but what if I want to use (S)CSS modules as well, and make them work with `@apply bg-black`?

**Reply by theodorusclarence** [reply]
I haven’t tested this, but you can try looking at the source if it doesn’t work

```js
    /**
     * Add *.scss support
     * @see https://nebulab.it/blog/nextjs-tailwind-storybook/
     */
    config.module.rules.push({
      test: /\.scss$/,
      use: ['style-loader', 'css-loader', 'postcss-loader', 'sass-loader'],
    });
```

**Reply by awthwathje** [reply]
You know what? Disregard this, it appears `@apply` syntax already works out of the box.

**Reply by theodorusclarence** [reply]
Nice, thanks for the confirmation

**Comment by D-Pagey** [comment]
Awesome, thanks a lot that all worked for me :)

**Comment by vivz753** [comment]
Nice guide. I ran into some errors parsing my .scss file because it had `@import` for a few things. 

Fixed by adding this into the `webpackFinal` of `.storybook/main.js`

```
    config.module.rules.push({
      test: /\.scss$/,
      use: ['style-loader', 'css-loader'],
    });
```

**Reply by vivz753** [reply]
That was just to get storybook running smoothly. Now that I have it up, I noticed my stories aren't displaying Tailwind CSS for some reason ): Will take another look tomorrow

**Reply by theodorusclarence** [reply]
I don't use scss, so I never tried this before. 
Kindly update to help the others. 🙌

**Reply by vivz753** [reply]
https://github.com/RyanClementsHax/storybook-addon-next worked right out of the box, thanks @volkyeth 

Having trouble loading next Images, however

**Comment by ali4heydari** [comment]
One other important thing is adding `-s public` parameter to `start-storybook` command in `package.json` file like bellow. otherwise static files like images not working.
Thanks for your great article.
```
"storybook": "start-storybook -s public -p 6006",
```

**Reply by theodorusclarence** [reply]
Hi, serving static files via Storybook CLI is now [deprecated](https://storybook.js.org/docs/react/configure/images-and-assets#%EF%B8%8F-deprecated-serving-static-files-via-storybook-cli) and changed with a configuration option `staticDirs: ['../public'],` in the `.storybook/main.js` 😄

**Reply by ali4heydari** [reply]
Oh, Thank you! 🙏🏻

**Comment by brunowowk** [comment]
Just wanted to point out to an alternative solution that requires less configuration and works seamlessly, with some extra integration goodies:
https://github.com/RyanClementsHax/storybook-addon-next

**Reply by theodorusclarence** [reply]
Thanks for the information, will check it out later!

**Comment by giwangdk** [comment]
Halo theo makasih article nya insightful banget
Mau tanya kalau misal mau setup untuk react Js sama kaya gini ya?

**Reply by theodorusclarence** [reply]
Hi, I have never tried using storybook with CRA because this tutorial is meant for Next.js. But I'm sure that there is a tutorial on google to use storybook with CRA.

**Comment by PawarAmit52** [comment]
Thanks for bringing this content for us as NextJS and Tailwind CSS are an amazing combination, both [possessing great UI Features](https://www.esparkinfo.com/top-react-js-ui-frameworks.html). NextJS as well as Tailwind CSS is being widely used for effective and appealing UI Solutions.

**Comment by fadlanzunima** [comment]
hi, what meaning of path in first line in storybook/main.js ?

**Reply by theodorusclarence** [reply]
it’s to resolve dirname path like this
`path.resolve(__dirname, '../src/')`

**Comment by jpvalery** [comment]
Hi there,

Thanks for sharing this. 

I'm running into the same issue as in https://github.com/storybookjs/storybook/issues/14405
<img width="763" alt="image" src="https://user-images.githubusercontent.com/4112343/174898524-cab9abc9-8509-490b-bf4d-a51edf1a2b39.png">

Any idea what could be causing this?

**Reply by ouedyan** [reply]
Hi. It is now way more easier to setup Storybook with Next.js. Check this Storybook addon : https://storybook.js.org/addons/storybook-addon-next.
And check out this template for an example : https://github.com/ouedyan/nextjs-template
